packages/editor/src/store/selectors.js

```diff
@@ -451,9 +479,17 @@ export function isEditedPostAutosaveable( state ) {
 		return true;
 	}
 
+	// To avoid an expensive content serialization, use the content dirtiness
+	// flag in place of content field comparison against the known autosave.
+	// This is not strictly accurate, and relies on a tolerance toward autosave
+	// request failures for unnecessary saves.
+	if ( hasChangedContent( state ) ) {
```

Comment:
I guess this is the fix for the performance issue. 

It might not be easy but I wonder if we can add an e2e test checking that we don't run `save` functions as we type :)

Reply:
> It might not be easy but I wonder if we can add an e2e test checking that we don't run `save` functions as we type :)

Hmm, might not be too hard! I'll give it a shot.